### DISCUSSION TITLE
How to wait for useTransition to finish animation before next component mount on animation?

### DISCUSSION THREAD

**Discussion by branlok** [pmndrs/react-spring#1450] [Support] [answered]
I notice the example for switching between component uses absolute position to overlap the animation. And I'm a little confuse about the interpolation bit  ```  opacity: opacity.to({ range: [0.0, 1.0], output: [0, 1] ```.  Is there an example to wait for animation to complete before mounting the next component?

**Comment by joshuaellis** [comment, marked as answer]
> Is there an example to wait for animation to complete before mounting the next component

No there's not... I'm not sure there's a 1:1 mapping for "wait till animation has finished" but there is a `trail` prop? Oh there's also an [`expires`](https://react-spring.io/hooks/use-transition#expires) prop

**Reply by joshuaellis** [reply]
Hey @branlok, did you manage to look into the above answer? If they didn't work, why not open a feature request and we can maybe look at implementing it! 🚀
